src/index.ts

```diff
@@ -738,7 +741,7 @@ export const union = <RTS extends Array<Mixed>>(
           pushAll(errors, validation.value)
         }
       }
-      return failures(errors)
+      return failures((fromArray(errors) as Some<Errors>).value)
```

Comment:
the cast is due to the early `return` inside the for loop. If we got to this point the `errors` array is already known to be non-empty.